### ISSUE TITLE
Failed to download ‘melpa’ archive.

### ISSUE THREAD

**Issue by jar398** [greghendershott/racket-mode#581]
I went to this page:
`https://www.racket-mode.com/#Configure-Emacs-to-use-MELPA`
which said:
`Type M-x package-refresh-contents RET . `
and when I do that (having also done the require and add-to-list above that) it says:

```
For information about GNU Emacs and the GNU system, type C-h C-a.
Importing package-keyring.gpg...done
Contacting host: elpa.gnu.org:443
Failed to download ‘gnu’ archive.
Failed to download ‘melpa-stable’ archive.
Contacting host: melpa.org:443
Package refresh done
Failed to download ‘melpa’ archive.
```

Please advise, and may I suggest you hone the documentation a little bit to anticipate people failures of this sort, or if this is a bug, please fix it...  and I am not ungrateful.

In case this is version skew: I'm using emacs 26 on debian something (PureOS). I used apt-get to install the debian racket package.

Jonathan

**Comment by sorawee** [comment]
Perhaps https://stackoverflow.com/a/58236306/718349 is relevant? In any case, I think this is a general Emacs issue, and not specific to Racket Mode.

**Comment by jar398** [comment]
You probably just did a web search for that error that I could have done, yes?  I am grateful, @sorawee .

The stackoverflow page is somewhere between funny and pathetic. 

> if I understand correctly ELPA has been broken since September, 2014 (at least) because some joker signed a package with a key that they didn't publish... And nobody seems to care

 is funny. I tried some of the suggested solutions and after running a maze of twisty little metaphorical passages (trying things out in emacs) conclude that the only one likely to work for me is to upgrade emacs to 26.3.  Unfortunately I choose not to do emacs upgrades outside of the official debian upgrade process (which is currently at 26.1). As a consequence I also choose not to use MELPA at this time...  hmm. I will try either scheme48 mode or DrRacket.  Or maybe I will look at racket-mode sources to see if there's a way to load racket-mode without using MELPA

Thanks

**Comment by stchang** [comment]
fwiw, I ran into a similar problem and what fixed it for me was running `M-x package-initialize` before `M-x package-refresh-contents`

Not sure if that's what is affecting you but the errors seem similar

(Enabling debug mode with `M-x toggle-debug-on-error` was also helpful for me, to get more detailed error messages)

**Comment by jar398** [comment]
That solution is in the SO discussion, but it didn't work for me.

Loading racket-mode from the git clone was both well documented and easy, so I'm in business. Now I'm getting a different problem. Will go RTFM now.

Thanks for your help.

**Comment by greghendershott** [comment]
I don't understand why you were getting download failures for both the ELPA and the MELPA Stable repos.

I notice the order of error messages makes it seem like your config might have been trying to get `melpa-stable` from `elpa.gnu.org`, which is wrong.

Furthermore Racket Mode isn't in `melpa-stable` -- instead it's in `melpa` (plain or "unstable").

The first step I document is [Configure Emacs to use MELPA](https://www.racket-mode.com/#Configure-Emacs-to-use-MELPA).

Anyway if OK with you're installing from source `master` branch, you're getting what ends up on `melpa` an hour or two later.

**Comment by jar398** [comment]
I did my best to follow the documentation. So yes I followed the configuration steps.

I am happy loading the files from a repo clone and the problem is fixed for my purposes. I think if I really wanted to pursue this I would set up a pristine Debian stable environment and follow the installation instructions.  If they don't work, you have instructions that don't apply to Debian stable (and they could be fixed to either say that, or to explain how to make it go).  If they do then there is something funny about my own setup (which would surprise me in a way because I think I am running a setup that should be identical to Debian stable for this purpose, but not surprise me because I am sometimes wrong).  This test shouldn't be too hard in the modern age of VMs and containers, but the task is not something I want to spend my time on given that my immediate problem is solved.
Best

**Comment by greghendershott** [comment]
Thanks for the feedback. To be clear, my intent was _not_ to say RTFM. :smile: I was responding to your suggestion I improve the documentation. Always fair; documentation always can be imrpoved. In this case I just wasn't sure exactly how/where to improve it, yet. But I'd be happy to follow up someday if/when we learn what could be said here. Cheers.

**Comment by greghendershott** [comment]
> documentation always can be imrpoved.

For example fixing spelling errors.

**Comment by jar398** [comment]
The documentation improvement I intended would have been a list of adequate prerequisites, with versions; then I could have seen that I might have been having trouble due to my having different versions of things; or at least I would have been alerted that there was no claim that the instructions would work in my case.
An additional step might be for the documentation to give potential remediations, such as what I eventually ended up doing (loading from the head of the repository).

The reason that I got a bit huffy is that I followed the instructions to the letter and they didn't work. One pacifies people like me by providing qualifiers like prerequisite version lists (i.e. setting expectations or disclaimers) and/or alternative approaches. Another thing one can say is "We have tested this release in the following configurations" hinting that there's been an effort toward portability but ultimately all bets are off if the target platform differs from what's known to work.

Thanks
